src/daemon/plugin.c

```diff
@@ -1078,14 +1084,96 @@ static int plugin_compare_read_func(const void *arg0, const void *arg1) {
     return 0;
 } /* int plugin_compare_read_func */
 
+/*
+ * Set the time of the first read according to the settings AlignRead and
+ * Interval in collectd.conf.
+ *
+ * The AlignRead value is in seconds. The value modulo 60 determines the second
+ * of a minute. The integer part of the value divided by 60 determines the
+ * minute of an hour. Optionally, decimal places can be used to set the
+ * fraction of a second (up to microseconds precision).
+ */
+static void plugin_set_first_read_time(read_func_t *rf) {
+  double alignRead = rf->rf_ctx.align_read; /* (default is -1.0) */
+
+  /* If AlignRead is not set, start reading immediately. */
+  if (alignRead < 0) {
+    rf->rf_next_read = cdtime();
+    return;
+  }
+
+  /* AlignRead values greater than Interval do not make sense. */
+  if (alignRead >= rf->rf_interval) {
```

Comment:
rf_interval is cdtime_t. You could not compare them directly.

Please do validation and verify corner cases!

Reply:
AlignRead is of type cdtime_t now (as you suggested).